### PR TITLE
TO golang -- add capability for custom changelog message per object type

### PR DESCRIPTION
**Pull request by dangogh**

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/api/change_log.go [hunk 1]
```diff
@@ -37,6 +37,10 @@ type ChangeLog struct {
 	LastUpdated tc.Time `json:"lastUpdated" db:"last_updated"`
 }
 
+type Logger interface {
+	ChangeLogMessage(action string, db *sqlx.DB) (string, error)
+}
+
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/api/change_log.go]: since it's the ChangeLog, what about

type ChangeLog Interface{
    Message{}
}

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1109/
Test PASSed.

File: traffic_ops/traffic_ops_golang/api/change_log.go (hunk 1 above)
Comment (DylanVolz) [reply traffic_ops/traffic_ops_golang/api/change_log.go]: I agree that Logger should be changed but should conform to the -er pattern.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1112/
Test PASSed.

File: traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go [hunk 2]
```diff
@@ -430,6 +430,20 @@ func (req *TODeliveryServiceRequest) Delete(db *sqlx.DB, user auth.CurrentUser)
 	return nil, tc.NoError
 }
 
+// ChangeLogMessage implements the api.ChangeLogger interface for a custom log message
+func (req *TODeliveryServiceRequest) ChangeLogMessage(action string, db *sqlx.DB) (string, error) {
```
Comment (DylanVolz) [inline traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go]: Why does ChangeLogMessage take a db handle as an argument? Will there be cases were constructing a struct's log message requires querying the db?

Comment (dangogh) [reply traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go]: possibly with foreign keys,  but the id probably would be enough..   we should discuss more..

**Comment by dangogh** [comment]
@DylanVolz removed db from the args -- no sense having it there with no known use case -- can be added later if we really need it..

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1113/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1117/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1130/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1131/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1132/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1134/
Test PASSed.

**Pull request merged**